stdlib/domain.ml

```diff
@@ -114,40 +114,49 @@ let self () = Raw.self ()
 
 module DLS = struct
 
-  type 'a key = int ref
+  type 'a key = int ref * (unit -> 'a)
 
-  type entry = {key: int ref; slot: Obj.t ref}
+  type entry = {key_id: int ref; slot: Obj.t ref}
```

Comment:
Perhaps I've asked this earlier, is there a reason why the type isn't

```
type entry = {key_id: int ref; mutable slot: Obj.t}
```

Reply:
`mutable` seems cleaner than `ref`. Changed this.